diesel/src/migrations/mod.rs

```diff
@@ -178,7 +179,7 @@ fn run_migrations<T, Conn: Connection>(conn: &Conn, migrations: T)
     -> Result<(), RunMigrationsError> where
         T: Iterator<Item=Box<Migration>>
 {
-    for migration in migrations {
+    for migration in migrations.into_iter().sorted_by(|a, b| Ord::cmp(&b.version(), &a.version())).into_iter().rev() {
```

Comment:
I'd rather not add a dependency for this. Let's just have this take a `Vec` instead of an iterator.